messaging/src/test/java/org/axonframework/queryhandling/StreamingQueryTest.java

```diff
@@ -0,0 +1,358 @@
+                    .expectErrorMatches(t -> t instanceof NoHandlerForQueryException
+                            && t.getMessage().startsWith("No suitable handler"))
+                    .verify();
+    }
+
+    @Test
+    void testThrottledFluxQuery() {
+        StreamingQueryMessage<String, Long> queryMessage =
+                new GenericStreamingQueryMessage<>("criteria",
+                                                   "throttledFluxQuery",
+                                                   Long.class);
+
+        StepVerifier.create(streamingQueryPayloads(queryMessage))
+                    .expectNext(0L, 1L, 2L, 3L, 4L, 5L, 6L, 7L)
+                    .verifyComplete();
+    }
+
+    @Test
+    void testBackpressureFluxQuery() {
+        StreamingQueryMessage<String, Long> queryMessage =
+                new GenericStreamingQueryMessage<>("criteria",
+                                                   "backPressure",
+                                                   Long.class);
+
+        StepVerifier.create(streamingQueryPayloads(queryMessage), 10L)
+                    .expectNextCount(10)
+                    .thenRequest(10)
+                    .expectNextCount(10)
+                    .thenCancel()
+                    .verify();
+    }
+
+    @Test
+    void testDispatchInterceptor() {
+        AtomicBoolean hasBeenCalled = new AtomicBoolean();
+
+        queryBus.registerDispatchInterceptor(messages -> {
+            hasBeenCalled.set(true);
+            return (i, m) -> m;
+        });
+
+        StreamingQueryMessage<String, String> queryMessage =
+                new GenericStreamingQueryMessage<>("criteria", "fluxQuery", String.class);
+
+        StepVerifier.create(streamingQueryPayloads(queryMessage))
+                    .expectNext("a", "b", "c", "d")
+                    .verifyComplete();
+
+        assertTrue(hasBeenCalled.get());
+    }
+
+    @Test
+    void testHandlerInterceptor() {
+        queryBus.registerHandlerInterceptor((unitOfWork, interceptorChain) ->
+                                                    ((Flux) interceptorChain.proceed()).map(it -> "a"));
+
+        StreamingQueryMessage<String, String> queryMessage =
+                new GenericStreamingQueryMessage<>("criteria", "fluxQuery", String.class);
+
+        StepVerifier.create(streamingQueryPayloads(queryMessage))
+                    .expectNext("a", "a", "a", "a")
+                    .verifyComplete();
+    }
+
+    @Test
+    void testErrorStream() {
+        StreamingQueryMessage<String, String> queryMessage =
+                new GenericStreamingQueryMessage<>("criteria", "errorStream", String.class);
+
+        StepVerifier.create(streamingQueryPayloads(queryMessage))
+                    .verifyErrorMatches(t -> t instanceof RuntimeException && t.getMessage().equals("oops"));
+    }
+
+    public static ConcurrentLinkedQueue<String> handlersInvoked = new ConcurrentLinkedQueue<>();
```

Comment:
Should this `static` field really be `public`? Furthermore, why is it somewhere in the middle of the test class?